src/qt3utils/nidaq/customcontrollers.py

```diff
@@ -0,0 +1,84 @@
+import nidaqmx
+import nipiezojenapy
+from typing import List
+import time
+import logging
+
+logger = logging.getLogger(__name__)
+class VControl(nipiezojenapy.BaseControl):
+
+    def __init__(self, device_name: str,
+                 write_channel: str = 'ao0',
+                 read_channel: str = None,
+                 scale_nm_per_volt: float = 8,
+                 move_settle_time: float = 0.001,
+                 min_position: float = 0.0,
+                 max_position: float = 80.0) -> None:
+        super().__init__()
+
+        self.device_name = device_name
+        self.write_channel = write_channel
+        self.read_channel = read_channel
+        self.scale_nm_per_volt = scale_nm_per_volt
+        self.minimum_allowed_position = min_position
+        self.maximum_allowed_position = max_position
+        self._settling_time_in_seconds = move_settle_time #10 millisecond settle time
+        self.last_write_value = None
+
+
+
+    def go_to_voltage(self, v: float = None) -> None:
+        '''
+        !//Sets the x,y,z position in microns.
+
+        //You do not need to specify all three axis values in order
+        to move in one direction. For example, you can call: go_to_position(z = 40)
+
+        raises ValueError if try to set position out of bounds.
+        '''
+
+        def goto(val):
```

Comment:
I realize you're following my lead here, but I think maybe this doesn't need to be an inner function. I was being a little lazy. Probably better to make this a private function of the class with a more verbose name, such as `_write_value_with_nidaqtask(val)` or other name if you think of something better.

Reply:
I don't even think it needs to be a private function--the functionality can just live in this function. It's not used anywhere else and it's not in a loop or anything.